### PR TITLE
[foonathan-memory]: update to 0.7-3

### PR DESCRIPTION
**Pull request by jiayuehua**
Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by Cheney-W — DISMISSED** [review]

**Comment by jiayuehua** [comment]
#30212

File: ports/foonathan-memory/vcpkg.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "foonathan-memory",
-  "version-date": "2019-07-21",
-  "port-version": 4,
+  "version": "0.7-3",
```
Comment (dan-shaw) [inline ports/foonathan-memory/vcpkg.json:3]: The `version` field needs to be dot separated sections. Maybe we can do `0.7.3` instead?

Comment (jiayuehua) [reply ports/foonathan-memory/vcpkg.json:3]: version change to 0.7.3, portfile's  https://github.com/jiayuehua/vcpkg/blob/38b57431b7f23d538b324e2fc009fbb22e570dc9/ports/foonathan-memory/portfile.cmake#LL8C1-L8C22 REF "v${VERSION}" change to v0.7-3 ? @Cheney-W  @dan-shaw

Comment (Cheney-W) [reply ports/foonathan-memory/vcpkg.json:3]: Yes, please modify the REF to `v0.7-3`.

Comment (jiayuehua) [reply ports/foonathan-memory/vcpkg.json]: have summit the change, @Cheney-W

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**